src/services/shortLink/test/index.test.ts

```diff
@@ -0,0 +1,552 @@
+        const response = await injectPost(app, {});
+        expect(response.statusCode).toEqual(StatusCodes.BAD_REQUEST);
+      });
+    });
+
+    describe('Not Found', () => {
+      it('Conflict if post short links with already exist alias', async () => {
+        const { item } = await mockPostData({ author: actor });
+        const { item: item2 } = await mockPostData({ author: actor });
+
+        await injectPost(app, mockShortLinkPayload({ item_id: item.id }));
+
+        const response = await injectPost(app, mockShortLinkPayload({ item_id: item2.id }));
+        expectException(response, new ShortLinkDuplication(MOCK_ALIAS));
+      });
+    });
+
+    describe('Not Found', () => {
+      it('Not found if post short links with invalid item id', async () => {
+        const response = await injectPost(app, mockShortLinkPayload());
+        expect(response.statusCode).toEqual(StatusCodes.NOT_FOUND);
+
+        const check = await injectGet(app, MOCK_ALIAS);
+        expect(check.statusCode).toEqual(StatusCodes.NOT_FOUND);
+      });
+    });
+
+    describe('Authorized Posts', () => {
+      afterEach(async () => {
+        const check = await injectGet(app, MOCK_ALIAS);
+        expect(check.statusCode).toEqual(StatusCodes.OK);
+        expect(check.json().alias).toEqual(MOCK_ALIAS);
+      });
+
+      it('Success if post short links with author member', async () => {
+        const { item } = await mockPostData({ author: actor });
+
+        const response = await injectPost(app, mockShortLinkPayload({ item_id: item.id }));
+        expect(response.statusCode).toEqual(StatusCodes.OK);
+        expect(response.json().alias).toEqual(MOCK_ALIAS);
+      });
+
+      it('Success if post short links with admin permission', async () => {
+        const { item } = await mockPostData({
+          author: bob,
+          member: actor,
+          permission: PermissionLevel.Admin,
+        });
+
+        const response = await injectPost(app, mockShortLinkPayload({ item_id: item.id }));
+        expect(response.statusCode).toEqual(StatusCodes.OK);
+        expect(response.json().alias).toEqual(MOCK_ALIAS);
+      });
+    });
+  });
+
+  describe('GET /short-links/:itemId', () => {
+    beforeEach(async () => {
+      ({ app, actor } = await build());
+      bob = await saveMember(BOB);
+    });
+
+    it('Not found if get short links with invalid item id', async () => {
+      const response = await injectGet(app, MOCK_ALIAS);
+      expect(response.statusCode).toEqual(StatusCodes.NOT_FOUND);
+    });
+
+    it('Success even if no connected', async () => {
+      const { item } = await mockPostData({ author: actor });
+      const payload = mockShortLinkPayload({ item_id: item.id, alias: MOCK_ALIAS });
+      const post = await injectPost(app, payload);
+      expect(post.statusCode).toEqual(StatusCodes.OK);
+      expect(post.json().alias).toEqual(MOCK_ALIAS);
+
+      logOut(app);
```

Comment:
After this, don't you need to resetAllMocks() in an afterEach hook ?

Reply:
Normally, there is a `jest.clearAllMocks();` in an `afterEach` at top of the test file.